### PR TITLE
feat: Extend AWS provider schema to support Fn::ToJsonString in environment variable

### PR DESCRIPTION
**Pull request by jurriaanpro**
Closes: #11409

### CODE REVIEW

**Comment by ronkorving** [comment]
This may be out of scope, and not something I really need myself. But just wanted to raise it as a topic of interest. Environment variables inevitably will not be the only place where this will be useful.

For example, when crafting a CloudWatch Dashboard under Resources, ToJsonString would allow people to use YAML to define a dashboard and have it convert to an unwieldy JSON string through this method.

```yaml
    rCloudWatchDashboardUx:
      Type: AWS::CloudWatch::Dashboard
      Properties:
        DashboardName: MyDashboard
        DashboardBody: !ToJsonString
          etc:
```

**Comment by jurriaanpro** [comment]
@ronkorving I think nothing is enforced under `resources` so long as you have a valid CloudFormation template. As mentioned in #11409, you'd need to include `Transform: AWS::LanguageExtensions` in your template as otherwise the `Fn::ToJsonString` won't be recognized/accepted.

Maybe someone that's deeper into the framework can confirm that.

**Comment by jurriaanpro** [comment]
@medikoo Coverage failed, but on code I didn't touch. Is there something I can do to fix this?

File: lib/plugins/aws/provider.js [hunk 1]
```diff
@@ -480,6 +480,25 @@ class AwsProvider {
...
+              'Fn::ToJsonString': {
+                oneOf: [
+                  { type: 'object' },
+                  { type: 'array' },
+                  { $ref: '#/definitions/awsCfFunction' },
+                  { $ref: '#/definitions/awsCfFindInMap' },
+                  { $ref: '#/definitions/awsCfGetAZs' },
+                  { $ref: '#/definitions/awsCfIf' },
+                  { $ref: '#/definitions/awsCfSelect' },
+                  { $ref: '#/definitions/awsCfSplit' },
```
Comment (medikoo) [inline lib/plugins/aws/provider.js:495]: Let's remove those definitions (it's fine to simply allow `object` and let follow CF validator with deep validation)

File: lib/plugins/aws/provider.js [hunk 2]
```diff
@@ -480,6 +480,25 @@ class AwsProvider {
             required: ['Fn::Sub'],
             additionalProperties: false,
           },
+          awsCfToJsonString: {
+            type: 'object',
+            properties: {
+              'Fn::ToJsonString': {
+                oneOf: [
```
Comment (medikoo) [inline lib/plugins/aws/provider.js:487]: Let's use `anyOf`

**Review by medikoo — COMMENTED** [review]
@jurriaanpro It looks good. Please see my comments

**Comment by jurriaanpro** [comment]
@medikoo I've applied the suggested changes. Could you check again?

File: lib/plugins/aws/provider.js [hunk 3]
```diff
@@ -480,6 +480,16 @@ class AwsProvider {
             required: ['Fn::Sub'],
             additionalProperties: false,
           },
+          awsCfToJsonString: {
+            type: 'object',
+            properties: {
+              'Fn::ToJsonString': {
+                anyOf: [{ type: 'object' }],
```
Comment (medikoo) [inline lib/plugins/aws/provider.js:487]: Let's drop `anyOf` wrapper (it doesn't make much sense to create `anyOf` with just one option)

**Review by medikoo — CHANGES_REQUESTED** [review]

File: lib/plugins/aws/provider.js (hunk 3 above)
Comment (jurriaanpro) [reply lib/plugins/aws/provider.js]: I think it actually needs to be 

```js
anyOf: [{ type: 'object' }, { type: 'array' }],
```

as both object an array are valid input. With only `{ type: 'object' }`, arrays aren't accepted.

I've adjusted it to that and tested with various `serverless.yaml` configs. Agree?

Comment (medikoo) [reply lib/plugins/aws/provider.js]: Sure that's great. Please re-request review once you feel it's ready

Comment (jurriaanpro) [reply lib/plugins/aws/provider.js]: PR is updated

**Review by medikoo — COMMENTED** [review]
@jurriaanpro can you rebase with main branch? I wonder why no github actions were run - feels as GitHub quirk

**Review by medikoo — APPROVED** [review]
Thank you!

**Pull request merged**